diff --git a/quickedit/includes/filter.inc b/quickedit/includes/filter.inc
index ca6f901..6c7e35d 100644
--- a/quickedit/includes/filter.inc
+++ b/quickedit/includes/filter.inc
@@ -83,6 +83,7 @@ function quickedit_filter_info_alter(&$info) {
     // http://drupal.org/project/mathjax: see http://drupal.org/node/1900436#comment-7198530
     // http://drupal.org/project/media
     'media_filter' => FILTER_TYPE_TRANSFORM_IRREVERSIBLE,
+    'media_filter_paragraph_fix' => FILTER_TYPE_TRANSFORM_IRREVERSIBLE,
     // https://drupal.org/project/node_embed
     'node_embed' => FILTER_TYPE_TRANSFORM_IRREVERSIBLE,
     // https://drupal.org/project/oembed
