diff --git a/quickedit.install b/quickedit.install
index 1bd69d7..6701b1d 100644
--- a/quickedit.install
+++ b/quickedit.install
@@ -149,7 +149,7 @@ function quickedit_requirements($phase) {
       $visible_field_count = 0;
       $node = entity_load_single('node', $node_id);
       foreach ($instances as $field_id => $instance) {
-        if (!empty($node->$field_id) && $instance['display']['default']['type'] != 'hidden') {
+        if (!empty($node->$field_id) && ((isset($instance['display']['full']) && $instance['display']['full']['type'] != 'hidden') || (!isset($instance['display']['full']) && $instance['display']['default']['type'] != 'hidden'))) {
           $visible_field_count++;
         }
       }
