diff --git a/maillog.module b/maillog.module
index e4107fe..11f1be8 100644
--- a/maillog.module
+++ b/maillog.module
@@ -121,6 +121,17 @@ function maillog_maillog_page($maillog) {
 function maillog_admin_settings() {
   $form = array();
 
+  $form['clear_maillog'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Clear Maillog'),
+  );
+
+  $form['clear_maillog']['clear'] = array(
+    '#type' => 'submit',
+    '#value' => t('Clear all maillog entries'),
+    '#submit' => array('maillog_clear_log'),
+  );
+
   $form['maillog_send'] = array(
     '#type' => 'checkbox',
     '#title' => t("Allow the e-mails to be sent."),
@@ -162,6 +173,11 @@ function maillog_views_api() {
   );
 }
 
+function maillog_clear_log() {
+  db_truncate('maillog')->execute();
+  drupal_set_message(t("All maillog entries deleted."));
+}
+
 /**
  * Implementation of hook_theme().
  */