diff --git a/node-edit-protection.js b/node-edit-protection.js
index 9b7d0a6..8749455 100755
--- a/node-edit-protection.js
+++ b/node-edit-protection.js
@@ -12,6 +12,10 @@
         });
       });

+      if ($(".node-form :input", context).hasClass("error")) {
+        edit = true;
+      }
+
       // Let all form submit buttons through
       $(".node-form input[type='submit'], .node-form button[type='submit']", context).each(function() {
         $(this).addClass('node-edit-protection-processed');
diff --git a/node_edit_protection.module b/node_edit_protection.module
index 2a52c36..e00d349 100644
--- a/node_edit_protection.module
+++ b/node_edit_protection.module
@@ -1,10 +1,25 @@
 <?php
 /**
- * @file
+ * @file
  * Stops page from changing when user is posting.
  */
+
+/**
+ * Implements hook_form_alter().
+ *
+ * Adds navigating away without saving protection to node edit forms.
+ */
 function node_edit_protection_form_alter(&$form, &$form_state, $form_id) {
-  if(isset($form['#node_edit_form'])) {
-    drupal_add_js(drupal_get_path('module', 'node_edit_protection') . '/node-edit-protection.js');
+  if (isset($form['#node_edit_form'])) {
+    // Calls function even after validation.
+    $form['#after_build'][] = 'node_edit_protection_after_build';
   }
 }
+
+/**
+ * Form #after_build callback for node_edit_protection_form_alter().
+ */
+function node_edit_protection_after_build($form, &$form_state) {
+  drupal_add_js(drupal_get_path('module', 'node_edit_protection') . '/node-edit-protection.js');
+  return $form;
+}