diff --git a/google_analytics/googleanalytics.module b/google_analytics/googleanalytics.module
index b45ee10..db30311 100755
--- a/google_analytics/googleanalytics.module
+++ b/google_analytics/googleanalytics.module
@@ -113,7 +113,7 @@ function googleanalytics_page_alter(&$page) {
   // 2. Track page views based on visibility value.
   // 3. Check if we should track the currently active user's role.
   // 4. Ignore pages visibility filter for 404 or 403 status codes.
-  if (preg_match('/^UA-\d+-\d+$/', $id) && (_googleanalytics_visibility_pages() || in_array($status, $trackable_status_codes)) && _googleanalytics_visibility_user($user)) {
+  if (preg_match('/^UA-\d+-\d+$/', $id) && (_googleanalytics_visibility_pages() || in_array($status, $trackable_status_codes))) {
 
     $debug = variable_get('googleanalytics_debug', 0);
     $url_custom = '';
