diff --git a/pdf_using_mpdf.module b/pdf_using_mpdf.module
index 3ec80db..2d3291c 100644
--- a/pdf_using_mpdf.module
+++ b/pdf_using_mpdf.module
@@ -95,7 +95,7 @@ function _pdf_using_mpdf_attributes_access($node) {
  * @param string $filename
  *   name of the PDF file to be generated.
  */
-function _pdf_using_mpdf_generator($html, $filename = NULL) {
+function _pdf_using_mpdf_generator($html, $filename = NULL, $node) {
 
   ini_set('Display_errors', 'On');
   error_reporting(E_ALL);
@@ -188,17 +188,19 @@ function _pdf_using_mpdf_generator($html, $filename = NULL) {
   // Set image DPI
   $mpdf->img_dpi = (int) variable_get('pdf_using_mpdf_img_dpi', 96);
 
+  global $language;
+
   // Enabling header option if available.
   $header = variable_get('pdf_using_mpdf_pdf_header');
   if (isset($header) && $header != NULL) {
-    $header = token_replace($header);
+    $header = token_replace($header, array('node' => $node), array('language' => $language));
     $mpdf->SetHTMLHeader($header);
   }
 
   // Enabling Footer option if available.
   $footer = variable_get('pdf_using_mpdf_pdf_footer');
   if (isset($footer) && $footer != NULL) {
-    $footer = token_replace($footer);
+    $footer = token_replace($footer, array('node' => $node), array('language' => $language));
     $mpdf->SetHTMLFooter($footer);
   }
 
@@ -347,13 +349,13 @@ function _pdf_using_mpdf_generator($html, $filename = NULL) {
  * @return bool
  *   TRUE if PDF is successfully generated and FALSE if it isn't.
  */
-function pdf_using_mpdf_api($html, $pdf_using_mpdf_pdf_filename = NULL) {
+function pdf_using_mpdf_api($html, $pdf_using_mpdf_pdf_filename = NULL, $node) {
   if (pdf_using_mpdf_library_exist() == TRUE) {
     if ($pdf_using_mpdf_pdf_filename === NULL) {
       $filename = explode(variable_get('pdf_using_mpdf_pdf_filename'), '[site:name]');
       $pdf_using_mpdf_pdf_filename = token_replace($filename[0]);
     }
-    _pdf_using_mpdf_generator($html, $pdf_using_mpdf_pdf_filename);
+    _pdf_using_mpdf_generator($html, $pdf_using_mpdf_pdf_filename, $node);
   }
   else {
     drupal_set_message(t('No mPDF Library Found in "sites/all/libraries" or "!default_module_path". Please download PHP mPDF PDF generation tool from <a href="http://www.mpdf1.com/">mPDF1.com</a>', array(
diff --git a/pdf_using_mpdf.pages.inc b/pdf_using_mpdf.pages.inc
index bc277f3..e00840e 100644
--- a/pdf_using_mpdf.pages.inc
+++ b/pdf_using_mpdf.pages.inc
@@ -38,7 +38,7 @@ function pdf_using_mpdf_generate_pdf($node) {
     $view = node_view($node, 'PDF');
     $html = drupal_render($view);
 
-    _pdf_using_mpdf_generator($html, $pdf_using_mpdf_pdf_filename);
+    _pdf_using_mpdf_generator($html, $pdf_using_mpdf_pdf_filename, $node);
   }
   else {
     drupal_set_message(t('mPDF library is not included. Please check your "sites/all/libraries" directory or " /sites/your-domain-name/libraries/ directory " or "!default_module_path" directory.', array(
