diff --git a/contextual-tabs.tpl.php b/contextual-tabs.tpl.php
index 0f6b221..76b1af9 100644
--- a/contextual-tabs.tpl.php
+++ b/contextual-tabs.tpl.php
@@ -21,12 +21,10 @@
   <?php endforeach; ?>
   <?php if (!empty($links)): ?>
     <li class="dropdown btn-group">
-      <a class="dropdown-toggle btn btn-small btn-sm btn-default" id="contextual-dropdown" data-toggle="dropdown" href="#" title="<?php print $title?>">
-        <i class="icon-cog"></i><span><?php print t('Configure');?></span>
+      <a class="dropdown-toggle btn btn-default" id="contextual-dropdown" data-toggle="dropdown" href="#" title="<?php print $title?>">
+        <i class="glyphicon glyphicon-cog"></i><span><?php print t('Configure');?></span>
       </a>
-      <ul class="dropdown-menu" role="menu" aria-labelledby="contextual-dropdown">
-        <li class="dropdown-column"><?php print $links; ?></li>
-      </ul>
+      <?php print $links; ?>
     </li>
   <?php endif; ?>
 </ul>
diff --git a/contextual_tabs.css b/contextual_tabs.css
index aaab4df..38b7b03 100644
--- a/contextual_tabs.css
+++ b/contextual_tabs.css
@@ -1,6 +1,5 @@
 #contextual-tabs {
   float: right;
-  margin-top: 23px;
 }
 
 #contextual-tabs > li {
diff --git a/contextual_tabs.module b/contextual_tabs.module
index 09958c9..f836f2b 100644
--- a/contextual_tabs.module
+++ b/contextual_tabs.module
@@ -89,7 +89,7 @@ function contextual_tabs_preprocess_page(&$variables) {
     $current_link = NULL;
     $links = array();
     $link_by_name = array();
-    $extract = explode("\n", variable_get('contextual_tabs_extract', ''));
+    $extract = preg_split('/\R/', variable_get('contextual_tabs_extract', ''));
 
     foreach ($tabs_links as $id => $link) {
       if ($link['href'] != $current) {
@@ -157,10 +157,13 @@ function contextual_tabs_preprocess_page(&$variables) {
       $build['links'] = theme('item_list', array(
         'items' => $data['links'],
         'type' => 'ul',
+        'attributes' => array(
+          'class' => 'dropdown-menu',
+          'aria-labelledby' => 'contextual-dropdown',
+        ),
       ));
     }
-    unset($variables['tabs']['#primary']);
+    //unset($variables['tabs']['#primary']);
     $variables['primarytabs'] = theme('contextual_tabs', $build);
   }
 }
-