diff --git a/libraries/http_request.inc b/libraries/http_request.inc
index 330fc4d..aca2c77 100644
--- a/libraries/http_request.inc
+++ b/libraries/http_request.inc
@@ -350,6 +350,22 @@ function feeds_http_request($url, array $options = array()) {
     }
   }
 
+$user_url = "https://www.projectdatasphere.org/projectdatasphere/html/rest/NCI/users";
+$request_url = "https://www.projectdatasphere.org/projectdatasphere/html/rest/NCI/requests";
+$request_other_approved_users_url = "https://www.projectdatasphere.org/projectdatasphere/html/rest/NCI/requests/otherapprovedusers";
+if($url == $user_url){
+$pdsdata = pds_data();
+$result->data = $pdsdata['user_data'];
+}
+if($url == $request_url){
+$pdsdata = pds_data();
+$result->data = $pdsdata['request_data_result'];
+}
+if($url == $request_other_approved_users_url){
+$pdsdata = pds_data();
+$result->data = $pdsdata['request_other_approved_users'];
+}
+
   // Set caches if asked.
   if ($options['cache_http_result']) {
     http_request_set_cache($url, $result);
